### DISCUSSION TITLE
鸿蒙4清软件后台会自动关闭搞快点的无障碍服务

### DISCUSSION THREAD

**Discussion by objzan** [gkd-kit/gkd#43] [Q&A]
鸿蒙4清软件后台会自动关闭搞快点的无障碍服务，每次清掉后台都要去无障碍再次打开，不知道其他手机系统会不会这样😂

**Comment by lisonge** [comment]
可以通过以下方式保持后台，以 MIUI14 为例

- 系统的最近任务界面-这个这个APP加锁
- 系统的本应用设置-省电策略 改为无限制
- 系统的本应用设置-允许自启动
- APP-设置-开启后台隐藏-然后返回桌面(不要上滑返回(home), 要正常返回(back))

**Reply by lisonge** [reply]
另外这个后台运行我理解的是软件是无法控制的，后台应用的存活控制完全在于系统以及用户的配置

我能做的就是开启前台通知，告诉系统我是一个需长时间运行的后台应用

并且我自己的红米k50u和红米k40，连续运行几天都没有被系统关闭后台

**Reply by ycdm** [reply]
我有一个建议，部分手机如果他总是清除后台，软件也无法控制，是否可以开放一个设置，打开该设置之后“启动没有界面的APP，或者瞬间关闭自己的Activity”，同时发出一个有通知栏的前台服务，用户如果真的要进入APP从通知栏进入可以打开界面。

然后让用户自己，借由手机自带的类似捷径或者自动任务的功能。
让用户设置反复定时启动APP，这样即无感，又能最大程度保活。

**Reply by lisonge** [reply]
APP 被关闭后，无障碍服务需要用户手动开启，你这种操作只是打开了 APP，并没有打开 无障碍服务

除非通过 shizuku 使用无障碍，只需要 APP 启动后即可，不需要用户手动打开 无障碍服务

**Reply by Teresian** [reply]
自启动应用需要手动关闭，然后三个选项全打开!
我倒是跳跳那边出现了杀后台后无法打开无障碍的问题。
[Screenshot_20231004_044059_com.huawei.systemmanager.jpg](https://github.com/gkd-kit/gkd/assets/133683433/b4fda893-2c32-4dcc-9864-ebdb13f83921)

**Reply by ycdm** [reply]
我认为 自动任务频繁自启应该可以减少被彻底关闭的情况，而且自启后检测 无障碍挂了也可以提示用户开启

**Comment by alinihao** [comment]
> 可以通过以下方式保持后台，以 MIUI14 为例
> 
> - 系统的最近任务界面-这个这个APP加锁
> - 系统的本应用设置-省电策略 改为无限制
> - 系统的本应用设置-允许自启动
> - APP-设置-开启后台隐藏-然后返回桌面(不要上滑返回(home), 要正常返回(back))
> 
> 
> 

我隐藏后台后，上滑返回无效来看看😂，侧滑返回可以了，建议把使用说明写在app里提示一下如何操作
回去再试试自己写规则，感谢作者
